src/pages/Restaurants.jsx

```diff
@@ -142,6 +30,23 @@ const CardDescription = ({ description, rating, priceRange, cuisine }) => {
 }
 
 export const RestaurantsPage = () => {
+  const [cardsData, setCardsData] = useState([])
+
+  const websitePort = window.location.port
+  const serverPort = '8080'
+  const url = window.location.href.replace(websitePort, serverPort)
```

Comment:
This not what we want.
Api endpoint is something that we always know. For example `GET /restaurants` and we can call this endpoint from any page or even any place in the codebase.

```suggestion
  const { restaurants } = useRestaurants()
```

Terefore please create a new hook that returns restaurants in src/hooks/useRestaurants.js:
```js
export const useRestaurants = (params) => {
  ...all the fetch logic here

  return { restaurants }
}
```

Reply:
Fixed